### PR TITLE
openlineage: add ClassVar to Airflow facets.

### PR DESCRIPTION
**Pull request by JDarDagran**
Recent OpenLineage release introduced a change that breaks mypy in main. This PR fixes it.

### CODE REVIEW

**Review by uranusjr — APPROVED** [review]

**Comment by JDarDagran** [comment]
I'm not sure why mypy fails now - it looks the opposite to what happens in main.

File: airflow/providers/openlineage/plugins/facets.py [hunk 1]
```diff
@@ -28,7 +30,7 @@ class AirflowMappedTaskRunFacet(BaseFacet):
     mapIndex: int
     operatorClass: str
 
-    _additional_skip_redact: list[str] = ["operatorClass"]
+    _additional_skip_redact: ClassVar[list[str]] = ["operatorClass"]
```
Comment (Taragolis) [inline airflow/providers/openlineage/plugins/facets.py]:
```suggestion
    _additional_skip_redact = ["operatorClass"]
```

File: airflow/providers/openlineage/plugins/facets.py [hunk 2]
```diff
@@ -63,7 +65,7 @@ class UnknownOperatorInstance(RedactMixin):
     properties: dict[str, object]
     type: str = "operator"
 
-    _skip_redact: list[str] = ["name", "type"]
+    _skip_redact: ClassVar[list[str]] = ["name", "type"]
```
Comment (Taragolis) [inline airflow/providers/openlineage/plugins/facets.py]:
```suggestion
    _skip_redact = ["name", "type"]
```

**Review by Taragolis — COMMENTED** [review]
You can't / shouldn't change parent class variable type: https://docs.python.org/3/library/typing.html#typing.ClassVar

That is the reason why `mypy` unhappy

**Comment by potiuk** [comment]
>  I'm not sure why mypy fails now - it looks the opposite to what happens in main.
How to check what versions of packages are installed in the CI env?

If PR does not change dependencies (setup.py/generated dependencies.json) then dependencies are taken from. Main constraints file. Those are the very same dependencies you have when you **just** build the image without --upgrade-to-newer-dependencies

**Comment by potiuk** [comment]
@JDarDagran https://github.com/apache/airflow/tree/constraints-main

**Comment by JDarDagran** [comment]
Okay, how about we remove annotations in provider now as suggested by @Taragolis?
Not sure if we ever want to add them back once contraints for `openlineage-python` are updated with pin to version higher than 1.5.0?

**Comment by potiuk** [comment]
Longer explanation:

> I'm not sure why mypy fails now - it looks the opposite to what happens in main.

Yes. This is entirely expected that your PR will have 'regular' constraints. Only the `main` build (also known as `canary` will attempt to upgrade dependencies to the newer ones. 

For 'regular` PRs the PR are using 'frozen' constraints which are automatically updated by the latest main build that passed all tests. This way when there is a change in released packages) - such as this one - all the regular PRs that do not touch dependencies will run using 'good' versions of packages.


This is all as designed as expected - this way change like this only fails canary build and a handful of PRs that change dependencies, but all 'standard' PRs are unaffected.

So yes - your PR uses the 'previous' version of openlineage libs - so you have to make your fix in the way that it is backwards compatible - and works for both - whAt was before the change and after. This is precisely what our users will have - some of them will use previous version of open lineage and some use the new version.

If you want to make a change that only works for newer version (which I think is not the case here) then you also have to update requirements for open lineage provider (and add >= x.y.z for example).

This way your PR will be 'changing' dependencies which means that it will automatically use 'upgrade to newer'


This is all precisely as designed and expected :)

**Comment by JDarDagran** [comment]
@potiuk thanks for your thorough explanation.
After all it seems like I should bump the dependency version as the change in `openlineage-python` is correct, we just want to have all new versions of OL provider to be compatible with this change. Is this reasoning correct?

**Review by potiuk — APPROVED** [review]

File: airflow/providers/openlineage/plugins/facets.py [hunk 3]
```diff
@@ -28,7 +30,9 @@ class AirflowMappedTaskRunFacet(BaseFacet):
     mapIndex: int
     operatorClass: str
 
-    _additional_skip_redact: list[str] = ["operatorClass"]
+    # openlineage-python<=1.5.0 does not annotate this as ClassVar
+    # mypy may complain about overriding instance variable with class variable
+    _additional_skip_redact: ClassVar[list[str]] = ["operatorClass"]  # type: ignore
```
Comment (potiuk) [inline airflow/providers/openlineage/plugins/facets.py]:
```suggestion
    _additional_skip_redact: ClassVar[list[str]] = ["operatorClass"]  # type: ignore[misc]
```

File: airflow/providers/openlineage/plugins/facets.py [hunk 4]
```diff
@@ -63,7 +67,9 @@ class UnknownOperatorInstance(RedactMixin):
     properties: dict[str, object]
     type: str = "operator"
 
-    _skip_redact: list[str] = ["name", "type"]
+    # openlineage-python<=1.5.0 does not annotate this as ClassVar
+    # mypy may complain about overriding instance variable with class variable
+    _skip_redact: ClassVar[list[str]] = ["name", "type"]  # type: ignore
```
Comment (potiuk) [inline airflow/providers/openlineage/plugins/facets.py]:
```suggestion
    _skip_redact: ClassVar[list[str]] = ["name", "type"]  # type: ignore[misc]
```

**Comment by potiuk** [comment]
BTw. @Taragolis in slack mentioned that just removing the annotation might work ... so maybe ?

**Comment by JDarDagran** [comment]
@potiuk  @Taragolis yeah, in the end we all agreed on removing the annotation. Went for it

**Review by potiuk — APPROVED** [review]
:D

**Review by Taragolis — APPROVED** [review]

**Pull request merged**